json_serializable/lib/src/enum_utils.dart

```diff
@@ -27,73 +27,78 @@ String? enumValueMapFromType(
     return null;
   }
 
-  MapEntry<FieldElement, dynamic> generateEntry(FieldElement fe) {
-    final annotation =
-        const TypeChecker.fromRuntime(JsonValue).firstAnnotationOfExact(fe);
-
-    dynamic fieldValue;
-    if (annotation == null) {
-      if (jsonEnum.valueField != null) {
-        // TODO: fieldRename is pointless here!!! At least log a warning!
-
-        final fieldElementType = fe.type.element2 as EnumElement;
-
-        final e = fieldElementType.getField(jsonEnum.valueField!);
-
-        if (e == null || e.isStatic) {
-          throw InvalidGenerationSourceError(
-            '`JsonEnum.valueField` was set to "${jsonEnum.valueField}", but '
-            'that is not a valid, instance field on '
-            '`${typeToCode(targetType)}`.',
-            element: targetType.element2,
-          );
-        }
-
-        final reader = ConstantReader(fe.computeConstantValue());
-        final valueReader = reader.read(jsonEnum.valueField!);
-        if (valueReader.validValueType) {
-          fieldValue = valueReader.literalValue;
-        } else {
-          throw InvalidGenerationSourceError(
-            '`JsonEnum.valueField` was set to "${jsonEnum.valueField}", but '
-            'that field does not have a type of String, int, or null.',
-            element: targetType.element2,
-          );
-        }
-      } else {
-        fieldValue = encodedFieldName(jsonEnum.fieldRename, fe.name);
-      }
-    } else {
-      final reader = ConstantReader(annotation);
+  final enumMap = {
+    for (var field in enumFields)
+      field: generateEntry(
+        field: field,
+        jsonEnum: jsonEnum,
+        targetType: targetType,
+      ),
+  };
+
+  final items = enumMap.entries
+      .map((e) => '  ${targetType.element2!.name}.${e.key.name}: '
+          '${jsonLiteralAsDart(e.value)},')
+      .join();
+
+  return 'const ${constMapName(targetType)} = {\n$items\n};';
+}
+
+Object? generateEntry({
```

Comment:
[nit] can this be private?

Reply:
Ugh, I need to remember to check the "request changes" box when I leave comments...